readthedocs/api/v2/views/model_views.py

```diff
@@ -306,12 +308,29 @@ def concurrent(self, request, **kwargs):
         }
         return Response(data)
 
+    def retrieve(self, *args, **kwargs):
+        """
+        Retrieves command data from storage.
 
-class BuildViewSet(SettingsOverrideObject):
-
-    """A pluggable class to allow for build cold storage."""
-
-    _default_class = BuildViewSetBase
+        This uses files from storage to get the JSON,
+        and replaces the ``commands`` part of the response data.
+        """
+        if not settings.RTD_SAVE_BUILD_COMMANDS_TO_STORAGE:
+            return super().retrieve(*args, **kwargs)
+
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        data = serializer.data
+        if instance.cold_storage:
+            storage = get_storage_class(settings.RTD_BUILD_COMMANDS_STORAGE)()
+            storage_path = '{date}/{id}.json'.format(
+                date=str(instance.date.date()),
+                id=instance.id,
+            )
+            if storage.exists(storage_path):
+                json_resp = storage.open(storage_path).read()
```

Comment:
We might want to add a try/except around the storage logic here, since I think it will likely fail more frequently than we want it to. Hopefully it isn't super slow.